python/packages/autogen-studio/autogenstudio/teammanager.py

```diff
@@ -1,50 +1,39 @@
-from typing import AsyncGenerator, Callable, Union, Optional
 import time
-from .database import ComponentFactory, Component
-from .datamodel import TeamResult, TaskResult, ComponentConfigInput
-from autogen_agentchat.messages import ChatMessage, AgentMessage
+from typing import AsyncGenerator, Callable, Optional, Union
+
+from autogen_agentchat.base._task import TaskResult
+from autogen_agentchat.messages import AgentMessage, ChatMessage
 from autogen_core.base import CancellationToken
 
+from .database import Component, ComponentFactory
+from .datamodel import ComponentConfigInput, TeamResult
+
 
 class TeamManager:
     def __init__(self) -> None:
         self.component_factory = ComponentFactory()
 
-    async def _create_team(
-        self,
-        team_config: ComponentConfigInput,
-        input_func: Optional[Callable] = None
-    ) -> Component:
+    async def _create_team(self, team_config: ComponentConfigInput, input_func: Optional[Callable] = None) -> Component:
         """Create team instance with common setup logic"""
-        return await self.component_factory.load(
-            team_config,
-            input_func=input_func
-        )
+        return await self.component_factory.load(team_config, input_func=input_func)
 
     def _create_result(self, task_result: TaskResult, start_time: float) -> TeamResult:
         """Create TeamResult with timing info"""
-        return TeamResult(
-            task_result=task_result,
-            usage="",
-            duration=time.time() - start_time
-        )
+        return TeamResult(task_result=task_result, usage="", duration=time.time() - start_time)
 
     async def run_stream(
         self,
         task: str,
         team_config: ComponentConfigInput,
         input_func: Optional[Callable] = None,
-        cancellation_token: Optional[CancellationToken] = None
+        cancellation_token: Optional[CancellationToken] = None,
     ) -> AsyncGenerator[Union[AgentMessage, ChatMessage, TaskResult], None]:
         """Stream the team's execution results"""
         start_time = time.time()
 
         try:
             team = await self._create_team(team_config, input_func)
-            stream = team.run_stream(
-                task=task,
-                cancellation_token=cancellation_token
-            )
+            stream = team.run_stream(task=task, cancellation_token=cancellation_token)
```

Comment:
The cancellation logic here needs to be updated once #4029 is resolved. The stream generator will get cancellation error which needs to be caught, rather than break.

Reply:
I agree. 
I have a proposal on how to fix this here - #4376 
Let me know your thoughts, once we align, I'll put in a PR.